docs/schema.rst

```diff
@@ -0,0 +1,124 @@
+.. with the License.  You may obtain a copy of the License at
+..
+..   http://www.apache.org/licenses/LICENSE-2.0
+..
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+..
+
+.. _schema-section:
+
+Pinot Schema
+============
+
+Pinot Schema consists of columns that can be categorized as dimensions, metric or time.
+
+**Dimensions** - These are columns that organize the data. For example, ``accountId``, ``country``, ``industry``, etc. These columns are used to slice/dice the data and typically appear in the ``selection``, ``filter`` and ``group-by`` sections in queries.
+
+**Metrics** - These are columns that represent quantative measurements. For example, ``numClicks``, ``pageViews``, etc. These columns typically appear in the aggregation section of the query, e.g., ``select sum(pageViews) from...``.
+
+**Time** - The time column represents the timestamp of the data. There is only one time column in a schema, and typically appears in the ``filter`` and ``group-by`` sections in queries. 
+
+A sample schema is shown below and will be used as an example to described the various fields.
+
+Sample Schema:
+~~~~~~~~~~~~~~
+
+.. code-block:: json
+
+   {
+     "schemaName": "flights",
+     "dimensionFieldSpecs": [
+       {
+         "name": "flightNumber",
+         "dataType": "LONG"
+       },
+       {
+         "name": "tags",
+         "dataType": "STRING",
+         "singleValueField": false
+       }
+     ],
+     "metricFieldSpecs": [
+       {
+         "name": "price",
+         "dataType": "DOUBLE"
+       }
+     ],
+     "timeFieldSpec": {
+       "incomingGranularitySpec": {
+         "name": "secondsSinceEpoch",
+         "dataType": "INT",
+         "timeFormat" : "EPOCH",
+         "timeType": "SECONDS"
+       },
+        "outgoingGranularitySpec": {
+         "name": "messageTimeHours",
+         "dataType": "INT",
+         "timeFormat" : "EPOCH",
+         "timeType": "DAYS"
+       }
+     }
+   }
+
+Schema Name:
+~~~~~~~~~~~~
+
+Every Pinot schema has a name that is used to identify it. For example, schema for a table is specified in the table conig by its name.
+
+
+FieldSpecs:
+~~~~~~~~~~~
+
+Each column in the schema is described using a ``fieldSpec`` that captures various attributes of the column, such as name, data-type, etc. A schema may contain an array of ``dimensionFieldSpecs`` that describe all the dimension columns, ``metricFieldSpecs`` that describe all the metric columns, and a ``timeFieldSpec`` that describes the time column.
+
+Dimensions:
+~~~~~~~~~~~~~~
+
+The schema example above has two dimensions ``flightNumber`` of type ``Long`` and ``tags`` of type ``String``. Data types supported for dimension columns are:
+
+* **INT**
```

Comment:
Use two backticks instead of two asterisks to make it similar to other pages. Not sure how these show up.

Reply:
The two backticks notations does not appear correct (in other pages as well). We should change it to bold by using this notations. Sample below: Note though, the backtick version in this comment looks much better than what we see in read-the-docs.

* ``INT``
* ``FLOAT``

v.s.

* **INT**
* **FLOAT**

Reply:
oh, then we need to standardize.